Microsoft.Toolkit.Uwp.UI.Controls.Primitives/StaggeredLayout/StaggeredLayout.cs

```diff
@@ -147,14 +147,16 @@ protected override Size MeasureOverride(VirtualizingLayoutContext context, Size
             double availableWidth = availableSize.Width;
             double availableHeight = availableSize.Height;
 
-            double columnWidth = Math.Min(DesiredColumnWidth, availableWidth);
+            // This ternary prevents the column width from being NaN, which would otherwise cause an exception when measuring item sizes
+            double columnWidth = double.IsNaN(DesiredColumnWidth) ? availableWidth : Math.Min(DesiredColumnWidth, availableWidth);
```

Comment:
I'm still trying to follow why the very similar logic works in `StaggeredPanel`  but not here in `StaggeredLayout`. @winston-de just curious if you followed the expressions to see where the failure occurred, just want to make sure we didn't overlook something else.

Reply:
The failure would occur [here](https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/09eb16132c13c2fadf98ae7e79fc754100330b4a/Microsoft.Toolkit.Uwp.UI.Controls.Primitives/StaggeredLayout/StaggeredLayout.cs#L200) with this error:
```
System.Runtime.InteropServices.COMException: 'Error HRESULT E_FAIL has been returned from a call to a COM component.'
```
All I know is that the exception occurs within the ``Measure`` function, There is no stack trace or other useful information.

The root cause of this issue could be a bug in WinUI, which could explain why it works in ``StaggeredLayout``?